packages/aws-cdk/lib/cdk-toolkit.ts

```diff
@@ -0,0 +1,101 @@
+import colors = require('colors/safe');
+import fs = require('fs-extra');
+import { format } from 'util';
+import { AppStacks, ExtendedStackSelection } from "./api/cxapp/stacks";
+import { IProvisioner } from './api/provisioner';
+import { printStackDiff } from './diff';
+import { deserializeStructure } from './serialize';
+
+export interface CdkToolkitProps {
+  /**
+   * The (stacks of the) CDK application
+   */
+  appStacks: AppStacks;
+
+  /**
+   * The provisioning engine used to apply changes to the cloud
+   */
+  provisioner: IProvisioner;
```

Comment:
We may have a different provisioner per artifact type. At the moment we only model "cloudformation stacks" as artifacts but we are on the process of generalizing this and then each artifact type could have a different provisioner, so I can envision this to be coupled with `AppStacks` (which will also be generalized).